gwt-jackson/src/main/java/com/github/nmorel/gwtjackson/client/deser/BaseDateJsonDeserializer.java

```diff
@@ -84,9 +85,8 @@ private SqlDateJsonDeserializer() { }
         }
 
         @Override
-        protected java.sql.Date deserializeString( String date, JsonDeserializerParameters params ) {
-            Date d = SQL_DATE_FORMAT.parse( date + " +0000" );
-            return new java.sql.Date( d.getTime() );
+        protected java.sql.Date deserializeString( String date, JsonDeserializationContext ctx, JsonDeserializerParameters params ) {
+            return new java.sql.Date( DateFormat.parse(ctx.isAdjustDatesToContextTimeZone(), SQL_DATE_FORMAT, date).getTime() );
```

Comment:
Maybe we could add a boolean to not check if the timezone exists in the pattern since we know it doesn't.

ie not call `hasTz` method for sql date